src/pybind/mgr/dashboard/controllers/auth.py

```diff
@@ -4,20 +4,29 @@
 import cherrypy
 import jwt
 
-from . import ApiController, RESTController
+from . import ApiController, RESTController, GroupDoc, EndpointDoc
 from .. import logger
 from ..exceptions import DashboardException
 from ..services.auth import AuthManager, JwtManager
 from ..services.access_control import UserDoesNotExist
 from ..services.sso import SSO_DB
 
 
+@GroupDoc(group="CephSpecialGroup", descr="DUMMY! Controller/group description")
 @ApiController('/auth', secure=False)
 class Auth(RESTController):
     """
     Provide authenticates and returns JWT token.
     """
-
+    @EndpointDoc(
+        descr="DUMMY! Endpoint description", 
+        param = {
+            'user': {
+                'username': (str, "Description of username"),
+                'password': (str, "Description of password")
```

Comment:
The reason we followed this approach was because docstrings don't enforce a strict grammar on how to write method documentation. Also, to document a REST API we require more information than just the description of the types of the method parameters and the return type.

These decorators capture what he need to document the REST API without the need to parse docstrings.

Reply:
It feels as if we're inventing a documentation scheme instead of re-using things that are already there. This isn't a big deal, until we want to automatically extract the docstrings with a different tool like sphinx for the official Ceph documentation. Using this custom style of method documentation will it hard to re-use it. 

Another thing is static type checking as in Python 3 using PEP 484. In addition to duplicating docstrings, we're also duplicating type annotations:
```python
In [1]: def f(x: int): pass

In [3]: f.__annotations__
Out[3]: {'x': int}
```

Reply:
In the future where we only need to support Python 3 we can try to make use of the static type information, but that does not cover all the cases we need to document a REST API. 

For instance, we need to be able to specify a different response JSON structure for each of the HTTP status codes.

Reply:
Another idea would be to go from the [OpenApi spec to Sphinx](https://sphinxcontrib-openapi.readthedocs.io/). This would solve the problem of having the need to document parameters in docstrings.